modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/NameSign.java

```diff
@@ -15,36 +15,44 @@
 import org.jetbrains.annotations.NotNull;
 
 public final class NameSign implements Listener {
-    String HEADER = "Name:";
+    private static final String HEADER = "Name:";
     @EventHandler
-    public void onCraftDetect(@NotNull CraftDetectEvent event){
+    public void onCraftDetect(@NotNull CraftDetectEvent event) {
         Craft c = event.getCraft();
         World w = c.getW();
-        for (MovecraftLocation location : event.getCraft().getHitBox()){
-            Block b = location.toBukkit(w).getBlock();
-            if (b.getType() == Material.SIGN_POST || b.getType() == Material.WALL_SIGN){
-                Sign s = (Sign) b.getState();
-                String name = "";
-                if (s.getLine(0).equalsIgnoreCase(HEADER)){
-                    if (s.getLine(1) != null){
-                        name += s.getLine(1) + " "; //reserved for prefix (HMA, SS, MS, USS, HMS, etc...)
-                    }
-                    if (s.getLine(2) != null){ // lines 2 and 3 are for the name
-                        name += s.getLine(2);
-                    }
-                    if (s.getLine(3) != null){
-                        name += s.getLine(3);
+        if(Settings.RequireNamePerm && event.getCraft().getNotificationPlayer() == null) {
+            return;
+        }
+        if(!Settings.RequireNamePerm || event.getCraft().getNotificationPlayer().hasPermission("movecraft.name.use")) {
```

Comment:
Try to avoid the [arrow anti pattern](http://wiki.c2.com/?ArrowAntiPattern) by flattening these if statements.